### DISCUSSION TITLE
Pack as a browser extension.

### DISCUSSION THREAD

**Discussion by Keritial** [qier222/YesPlayMusic#1394] [Ideas]
As title, can it be packed as a extension to use it directly in a browser(like Chrome and Firefox) without downloading the huge Electron framework(basicly another browser) or serving it and the API it uses on a server?

**Comment by memorydream** [comment]
readme中提供了一个在浏览器中运行的demo链接
ypm支持PWA
你可以自行构建它并host到自己的服务器上，readme中也提供了如何部署的步骤指导
web版缺少Unblock Netease Music相关功能，因为unm与ypm通过ipc进行通信，这些需要在node环境中使用

---

虽然可能有些文不对题，但是只是想说，ypm可以脱离electron运行，虽然不是以浏览器扩展的形式

**Reply by pan93412** [reply]
**Translation**

We've provided a demo instance that can run in browsers in our README file.

As YPM supports PWA, you can manually build and host it on your server. We've also written the instruction to deploy it in our README file.

The Web edition lacks the UnblockNeteaseMusic function, as YPM communicates UNM with IPC, and IPC can only work in the Node.js environment.

---

Sorry for not being at issue. I just want to explain that YPM can run without Electron, though it is not a browser extension.

**Comment by Keritial** [comment]
Thanks, although it's not what I mean, I'll figure it out myself.
